db/pebble/db.go

```diff
@@ -29,38 +36,51 @@ type DB struct {
 	pebble   *pebble.DB
 	wMutex   *sync.Mutex
 	listener db.EventListener
-}
 
-// New opens a new database at the given path with default options
-func New(path string) (db.DB, error) {
-	return newPebble(path, nil)
+	compTimeMeter        prometheus.Counter // Total time spent in database compaction
+	compReadMeter        prometheus.Counter // Total bytes read during compaction
+	compWriteMeter       prometheus.Counter // Total bytes written during compaction
+	writeDelayCountMeter prometheus.Counter // Total write delay counts due to database compaction
+	writeDelayTimeMeter  prometheus.Counter // Total write delay duration due to database compaction
+	diskSizeGauge        prometheus.Gauge   // Tracks the size of all of the levels of the database
+	diskWriteMeter       prometheus.Counter // Measures the effective amount of data written to disk
+	memCompGauge         prometheus.Gauge   // Tracks the amount of memory allocated for compaction
+	level0CompGauge      prometheus.Gauge   // Tracks the number of level-zero compactions
+	nonlevel0CompGauge   prometheus.Gauge   // Tracks the number of non level-zero compactions
+	seekCompGauge        prometheus.Gauge   // Tracks the number of table compaction caused by read opt
+	manualMemAllocGauge  prometheus.Gauge   // Tracks the amount of non-managed memory currently allocated
+
+	activeComp          int           // Current number of active compactions
+	compStartTime       time.Time     // The start time of the earliest currently-active compaction
+	compTime            atomic.Int64  // Total time spent in compaction in ns
+	level0Comp          atomic.Uint32 // Total number of level-zero compactions
+	nonLevel0Comp       atomic.Uint32 // Total number of non level-zero compactions
+	writeDelayStartTime time.Time     // The start time of the latest write stall
+	writeDelayCount     atomic.Int64  // Total number of write stall counts
+	writeDelayTime      atomic.Int64  // Total time spent in write stalls
```

Comment:
Can we add one or two comments just to help categorise these?

// Compaction Metrics (Prometheus Counters and Gauges)
	compTimeMeter        prometheus.Counter // Total time spent in database compaction
	compReadMeter        prometheus.Counter // Total bytes read during compaction
	compWriteMeter       prometheus.Counter // Total bytes written during compaction
	level0CompGauge      prometheus.Gauge   // Tracks the number of level-zero compactions
	nonlevel0CompGauge   prometheus.Gauge   // Tracks the number of non level-zero compactions
	seekCompGauge        prometheus.Gauge   // Tracks the number of table compactions caused by read operations

	// Write Delay Metrics (Prometheus Counters and Gauges)
	writeDelayCountMeter prometheus.Counter // Total write delay counts due to database compaction
	writeDelayTimeMeter  prometheus.Counter // Total write delay duration due to database compaction

	// Disk and Memory Usage Metrics (Prometheus Counters and Gauges)
	diskSizeGauge        prometheus.Gauge   // Tracks the size of all database levels (disk usage)
	diskWriteMeter       prometheus.Counter // Measures the effective amount of data written to disk
	memCompGauge         prometheus.Gauge   // Tracks the amount of memory allocated for compaction
	manualMemAllocGauge  prometheus.Gauge   // Tracks the amount of non-managed memory currently allocated

	// Operational Counters for Compaction (Atomic Variables and Others)
	activeComp          int           // Current number of active compactions
	compStartTime       time.Time     // The start time of the earliest currently-active compaction
	compTime            atomic.Int64  // Total time spent in compaction (in nanoseconds)
	level0Comp          atomic.Uint32 // Total number of level-zero compactions
	nonLevel0Comp       atomic.Uint32 // Total number of non level-zero compactions

	// Write Delay Operational Counters (Atomic Variables)
	writeDelayStartTime time.Time     // The start time of the latest write stall due to compaction
	writeDelayCount     atomic.Int64  // Total number of write stalls
	writeDelayTime      atomic.Int64  // Total time spent in write stalls

Reply:
nice :) makes it much cleaner